### ISSUE TITLE
How to deal with challenge required?

### ISSUE THREAD

**Issue by spapapan** [ErrorxCode/EasyInsta#14]
I try to request the followers of a user like this:

        instagram.profile().getFollowers("user1")
                .setOnCompleteCallback(call ->
                {
                    if (call.isSuccessful)
                    {
                        List<String> followers = call.result;
                        for (String follower : followers)
                        {
                            Print.e(follower);
                        }
                    }
                });

But I get the error challenge_required. How to deal with this error?

**Comment by spapapan** [comment]
It is worth noting that I got this error the second time I tried to make a request, so it's not that I spammed.

**Comment by ErrorxCode** [comment]
**A big problem nowadays**. Looks like instagram is improving their security algorithm day by day. They are somehow detection that we are using their API in wrong way.

To avoid these challenge, you can :-
- Use cache instead of logging in everytime
- Add human interpreted gap between every request
- Do not use new accounts. accounts which are not older then 90 days suffer this problem the most. Instagram keep tracks of new account.


_I will update it as soon as I get the fix_

**Comment by noirtier-villefort** [comment]
> **A big problem nowadays**. Looks like instagram is improving their security algorithm day by day. They are somehow detection that we are using their API in unauthenticated way
> 
> To avoid these challenge, you can :-
> 
>     * Use cache instead of logging everytime
> 
>     * Add human interpreted gap between every request
> 
>     * Do not use new accounts. accounts which are not older then 90 days suffer this problem the most. Instagram keep tracks of new account.
> 
> 
> _I will update it as soon as I get the fix_

Maybe we can avoid it by updating APP_VERSION in instagram4j IGConstants and using new IGAndroidDevice`s ???

**Comment by ErrorxCode** [comment]
> > **A big problem nowadays**. Looks like instagram is improving their security algorithm day by day. They are somehow detection that we are using their API in unauthenticated way
> > To avoid these challenge, you can :-
> > ```
> > * Use cache instead of logging everytime
> > 
> > * Add human interpreted gap between every request
> > 
> > * Do not use new accounts. accounts which are not older then 90 days suffer this problem the most. Instagram keep tracks of new account.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > _I will update it as soon as I get the fix_
> 
> Maybe we can avoid it by updating APP_VERSION in instagram4j [IGConstants](https://github.com/instagram4j/instagram4j/blob/39635974c391e21a322ab3294275df99d7f75f84/src/main/java/com/github/instagram4j/instagram4j/IGConstants.java#L12) and using new [IGAndroidDevice](https://github.com/instagram4j/instagram4j/blob/39635974c391e21a322ab3294275df99d7f75f84/src/main/java/com/github/instagram4j/instagram4j/IGAndroidDevice.java#L7)`s ???

Yeah, this is one of the crucial factor which is causing this problem. But instead of fixing that, we are again reverse engineering Instagram API with all the factors and constants taking in account. This one will be using Web API where as old **instagran4j** was using mobile version API.
So instead of contributing to **instagram4j**, you can [contribute in making of this new API](https://github.com/ErrorxCode/EasyInsta#contributing)
